edi_oca/models/edi_backend.py

```diff
@@ -670,3 +674,41 @@ def _is_valid_edi_action(self, action, raise_if_not=False):
             if raise_if_not:
                 raise
             return False
+
+    def _check_obsolete(self, exchange_record):
```

Comment:
A check should never write on the record IMO.
As for this check, it won't be needed if the state changes to `obsolete` before.
I propose a different solution on the exc record. See below.